src/libraries/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs

```diff
@@ -3070,5 +3070,26 @@ private static unsafe byte[] ObfuscateWithExtraData(byte[] unobfuscated, bool se
 
             return obfuscated;
         }
+
+        [Fact]
+        public void GetAssemblyName()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("assemblyFile", () => MetadataReader.GetAssemblyName(null));
+            AssertExtensions.Throws<ArgumentException>("path", null, () => MetadataReader.GetAssemblyName(string.Empty));
+            Assert.Throws<FileNotFoundException>(() => MetadataReader.GetAssemblyName("IDontExist"));
+
+            using (var tempFile = new TempFile(Path.GetTempFileName(), 0)) // Zero-size file
+            {
+                Assert.Throws<BadImageFormatException>(() => MetadataReader.GetAssemblyName(tempFile.Path));
+            }
+
+            using (var tempFile = new TempFile(Path.GetTempFileName(), 42))
+            {
+                Assert.Throws<BadImageFormatException>(() => MetadataReader.GetAssemblyName(tempFile.Path));
+            }
+
+            Assembly a = typeof(MetadataReaderTests).Assembly;
+            Assert.Equal(new AssemblyName(a.FullName).ToString(), MetadataReader.GetAssemblyName(AssemblyPathHelper.GetAssemblyLocation(a)).ToString());
```

Comment:
This is just copied from `AssemblyNameTest` as the logic is moved into this project assumed the test also should be moved, though did not removed original one in `AssemblyNameTest` as `AssemblyName` still had the public API https://github.com/dotnet/runtime/blob/8cd9d63693d75a526ddd72b372ffc12cc2ab3735/src/libraries/System.Reflection/tests/AssemblyNameTests.cs#L295-L317
Please let me know if you don't want to keep duplicate test and suggest to remove one of them

Reply:
We can make all tests to call both APIs.

Reply:
Yeah that could work, the reason of keeping it in each project because folks mostly run individual project tests only, rarely run all tests. So that they could detect the issue before raising a PR